### PR TITLE
Genetics Fixes and Tweaks

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Changes
- Can have hulk and dwarf at the same time
- Can have Heat and Cold resist at the same time
 - This was meant to be allowed provided it passed some convoluted RNG check...in practice, that's never the case and doesn't work
 - Adds overlay for having both heat+cold resist
- Didn't allow for chameleon+Cloak of Darkness together because it has extremely odd behavior
- I wanted to make polymorph change species, but I knew people would bitch to no end about it
- Changes some flavortexts when you acquire genes so it's more obvious what they do
- Ensures all genes have a message on activation/deactivation (minus monkey, of course).
- Remote View can only be used to spy on those who who also have remote view
 - This is probably the single most broken genetics ability that exists in the entire game for hunting people down

Fixes
- Fixes mitochondria offering a cast list when it's always meant to cast on yourself
- Fixes cold mutation not having an overlay
- Fixes hallucinations ticking down and double their rates and creating twice as many hallucinations as necessary

Code Cleanup
- Removes commented out coded
- Removes unused stability var
- Removes the random healing from Xenos (forgot to remove this in a previous PR that removed it for humans)
- Removes colon `:` usage.

Bugs I'm Aware of, but Don't really want to fix:
- The resist fire block is technically supposed to prevent fire damage---this definitely applies to non-human mobs, but is bugged out for human mobs.

:cl: Fox McCloud
tweak: Can have hulk+dwarf mutations together
tweak: Can have heat+cold resist together
tweak: Can only remotely view people who also have remote view
fix: Fixes cold mutation not having an overlay
fix: Hallucinations will no longer tick down twice as fast as they should nor will they spawn twice as many hallucinations as they should
fix: Incendiary mitochondria no longer asks who you want to cast it on (it always is meant to be cast on yourself)
/:cl:

### CODE REVIEW

**Comment by ProperPants** [comment]
>Can have hulk and dwarf at the same time
2fd2b11524acd526d312972e7d23b2d9

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 1]
```diff
@@ -120,10 +120,6 @@
 	return
 
 /mob/living/carbon/alien/handle_mutations_and_radiation()
-	if(getFireLoss())
-		if((RESIST_HEAT in mutations) || prob(5))
-			adjustFireLoss(-1)
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/alien/alien.dm]: How exactly do xenomorphs remove burn damage, now? Is there something with weeds? They don't have healing items.

**Comment by tigercat2000** [comment]
``` The resist fire block is technically supposed to prevent fire damage---this definitely applies to non-human mobs, but is bugged out for human mobs. ```
You could fix this pretty easily by checking if the fire block is active in [this](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/living/carbon/human/life.dm#L415-L433) calculation. How you check for active blocks, I'm not entirely sure.

File: code/modules/mob/living/carbon/alien/alien.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/alien/alien.dm]: https://github.com/ParadiseSS13/Paradise/blob/49818d587da8495ac62bdbcbf0196431a44a924b/code/modules/surgery/organs/subtypes/xenos.dm#L87

**Comment by Fox-McCloud** [comment]
@tigercat2000 

When I say fire damage, I mean *all* fire damage: https://github.com/ParadiseSS13/Paradise/blob/49818d587da8495ac62bdbcbf0196431a44a924b/code/modules/mob/living/damage_procs.dm#L18

**Comment by tigercat2000** [comment]
@Fox-McCloud that's handled here https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/living/carbon/human/human_damage.dm#L392-L399

**Comment by Fox-McCloud** [comment]
I know

**Comment by QQCucumber** [comment]
How about a cancel button for the block selector menu for block injector?

**Comment by QQCucumber** [comment]
Also, since remote view is getting nerfed to only include other people that are remotely viewed, could it receive a small buff in being able to see emotes and such through other people's eyes? It never made sense that you couldn't tell what they're doing.

**Comment by Fox-McCloud** [comment]
>Also, since remote view is getting nerfed to only include other people that are remotely viewed, could it changed to be able to see emotes and such through other people's eyes? It never made sense that you couldn't tell what they're doing.

You're viewing them, your ears aren't at their location.

**Comment by QQCucumber** [comment]
>>You're viewing them, your ears aren't at their location.

That's why I said emotes.

**Comment by tigercat2000** [comment]
@Fox-McCloud Though, that would be kinda cool too. Given that it's so limited, it would be nice if you could actually just observe your ~~metabuddies~~ genetics partner completely.

**Pull request merged**